lib/rspec/matchers/built_in/match.rb

```diff
@@ -5,6 +5,22 @@ module BuiltIn
       # Provides the implementation for `match`.
       # Not intended to be instantiated directly.
       class Match < BaseMatcher
+        # Used to specify the captures we match against
+        # Accepts variadic non-named captures
+        # @return [self]
+        def with_captures(*captures)
+          @expected_captures = captures
+          extend WithCaptures
+        end
```

Comment:
> Whereas doing with_captures on a string/string assertion isn't even a valid predicate.
Yeah but at 'match' time we only know that expected should have been a regular expression, consider the case where a string is accidentally returned instead of a regular expression, it would make sense to use `with_captures` but have it fail because it doesn't match....